homeassistant/components/airgradient/sensor.py

```diff
@@ -0,0 +1,139 @@
+
+
+@dataclass(frozen=True, kw_only=True)
+class AirGradientSensorEntityDescription(SensorEntityDescription):
+    """Describes AirGradient sensor entity."""
+
+    value_fn: Callable[[Measures], StateType]
+
+
+SENSOR_TYPES: tuple[AirGradientSensorEntityDescription, ...] = (
+    AirGradientSensorEntityDescription(
+        key="pm01",
+        device_class=SensorDeviceClass.PM1,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.pm01,
+    ),
+    AirGradientSensorEntityDescription(
+        key="pm02",
+        device_class=SensorDeviceClass.PM25,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.pm02,
+    ),
+    AirGradientSensorEntityDescription(
+        key="pm10",
+        device_class=SensorDeviceClass.PM10,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.pm10,
+    ),
+    AirGradientSensorEntityDescription(
+        key="temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.ambient_temperature,
+    ),
+    AirGradientSensorEntityDescription(
+        key="humidity",
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.relative_humidity,
+    ),
+    AirGradientSensorEntityDescription(
+        key="signal_strength",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.signal_strength,
+    ),
+    AirGradientSensorEntityDescription(
+        key="tvoc",
+        translation_key="total_volatile_organic_component_index",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.total_volatile_organic_component_index,
+    ),
+    AirGradientSensorEntityDescription(
+        key="nitrogen_index",
+        translation_key="nitrogen_index",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda status: status.nitrogen_index,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up AirGradient sensor entities based on a config entry."""
+
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        AirGradientSensor(coordinator, description) for description in SENSOR_TYPES
```

Comment:
All sensor measurements [can be `None`](https://github.com/airgradienthq/python-airgradient/blob/main/src/airgradient/models.py#L19-L43). I assume this means that they are not reported by the device. In that case, you should not add entities for them.

Reply:
They can be None if the device just booted. I am still informing if every measurement is present on every device